src/joint/joint.rs

```diff
@@ -125,6 +125,43 @@ where
         self.world_velocity_cache.replace(None);
         Ok(())
     }
+    /// Set the clamped position of the joint
+    ///
+    /// It refers to the joint limit and clamps the argument. This function does nothing if this is fixed joint.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// extern crate nalgebra as na;
+    /// extern crate k;
+    ///
+    /// // Create rotational joint with Y-axis
+    /// let mut rot = k::Joint::<f64>::new("r0", k::JointType::Rotational { axis: na::Vector3::y_axis() });
+    ///
+    /// let limits = k::joint::Range::new(-1.0, 1.0);
+    /// rot.limits = Some(limits);
+    ///
+    /// // Initial position is 0.0
+    /// assert_eq!(rot.joint_position().unwrap(), 0.0);
+    /// rot.set_joint_position_clamped(2.0);
+    // assert_eq!(rot.joint_position().unwrap(), 1.0);
+    /// rot.set_joint_position_clamped(-2.0);
+    /// assert_eq!(rot.joint_position().unwrap(), -1.0);
+    /// ```
+    ///
+    pub fn set_joint_position_clamped(&mut self, position: T) {
+        if let JointType::Fixed = self.joint_type {
```

Comment:
Please use is_movable(), and remove unnecessary `else` (L155).

```rust
if !self.is_movable() {
  return;
}
```

Reply:
Ok. I found that `set_joint_position` function also use such a if-statement:
https://github.com/OTL/k/blob/b373b6e2c841a7c033685805c8699c8998129ad3/src/joint/joint.rs#L107

I've replaced both of them by `is_movable`.